### ISSUE TITLE
provide high-level API

### ISSUE THREAD

**Issue by ubnt-intrepid** [ubnt-intrepid/polyfuse#39]
The current library implementation focuses on porting the libfuse's low-level API (described in `fuse_lowlevel.h`). and the convenience from the perspective of the user's side of the filesystem implementors is not considered.

In order to grow this project as a more use-friendly FUSE framework, it is needed to hide the protocol details and provide the higher-level API for filesystem authors (here, the term "high-level" is different from what used in `libfuse` and it is not a path-based API).

**Comment by ubnt-intrepid** [comment]
See also: https://github.com/hanwen/go-fuse/tree/master/fs

**Comment by jamadazi** [comment]
Yes, I am currently trying to learn polyfuse-tokio and implement a sysfs/debugfs inspired pseudo fs for a tokio-based daemon I am writing, to allow controlling it at runtime using regular shell commands. As I am going through the examples to learn the API, it feels like there would be a lot of potential for crating some nice rusty abstractions (possibly with proc macros) for such use cases.

If it was easy to create such pseudo-filesystems to expose the internals of a program, it could become a very convenient way of adding a runtime control interface to tokio-based applications.
